### PR TITLE
Setting up cli interactive flow

### LINKED ISSUE #70: Create interactive publish function
Create a function with multiple steps that helps you publish an asset

ex:
1. what themplate do you want to use 1,2,4?
2. What is the name of the asset
3.  ...

We can get inspired from the current publish screen in the market https://market.oceanprotocol.com/publish/1

### PR DESCRIPTION
**Pull request by jamiehewitt15**
Fixes #70 

Changes proposed in this PR:

- Setting up cli interactive flow
- Publishing asset
- Adding interactive flow to the system tests.

## To test (with barge):
1. run barge locally
2. run ocean node locally
3. set the envs in ocean-cli (`export NODE_URL="http://127.0.0.1:8001"`). No need to set Aquarius or provider URL anymore.
4. `npm run build && npm run cli start`

## To test (with Oasis testnet):
1. get rose tokens on the testnet using the faucet: https://faucet.testnet.oasis.dev
2. Run the node with oasis tesnet in the RPC env: `
export RPCS='{ "23295":{ "rpc":"https://testnet.sapphire.oasis.io", "chainId": 23295, "network": "sapphire-testnet", "chunkSize": 1000 }}'`
3. set the envs in ocean-cli (`export NODE_URL="http://127.0.0.1:8001"`). No need to set Aquarius or provider URL anymore.
4. `npm run build && npm run cli start`

### CODE REVIEW

**Comment by mihaisc** [comment]
can you do a quick update on the environment vars, so we don't have provider, aquarius

File: src/helpers.ts [hunk 1]
```diff
@@ -197,6 +200,179 @@ export async function createAsset(
 	return ddo.id;
 }
 
+
+export async function createSapphireAsset(
```
[the remark this answers is not in the document]
Comment (jamiehewitt15) [reply src/helpers.ts:204]: Cool yeah, good point. I'll change it

[the remark this answers is not in the document]
Comment (paulo-ocean) [reply src/helpers.ts:204]: for instance; i had something like this (replacing "old" createAsset from cli):
```
const assetIDNewImpl = await createAsset(
				asset.nft.name,
				asset.nft.symbol,
				this.signer,
				asset.services[0].files,
				4, // template 4 
				asset,
				encryptDDO,
				this.providerUrl,
				null,
				this.aquarius
			)
```
(i had the template id 4 harcoded )

**Review by paulo-ocean — APPROVED** [review]
Lgtm. 
One small comment, as we are not using the `createAsset` from SDK
This part:
```
ddo.services[0].files = await ProviderInstance.encrypt(
		assetUrl,
		chainId,
		macOsProviderUrl || providerUrl
	);
```
If the templateIndex is 4
we need to remove the files object from the ddo service (actually we need to set it to empty string, not null, due to schema validations)
Like this
```
if(templateIndex === 4){ 
  ddo.services[0].files = ''
 } else ...
```

**Comment by jamiehewitt15** [comment]
> we need to remove the files object from the ddo service (actually we need to set it to empty string, not null, due to schema validations)

Ok, thanks for letting me know. I'll adjust that.

**Comment by jamiehewitt15** [comment]
@paulo-ocean I've updated it to set the files as an empty string for template index 4. Let me know if it's alright and I'll merge this

**Comment by paulo-ocean** [comment]
> @paulo-ocean I've updated it to set the files as an empty string for template index 4. Let me know if it's alright and I'll merge this

go for it

**Pull request merged**